File: MedicApp/Models/Roles.cs
namespace MedicApp.Models
{
    public class Roles
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public bool IsDeleted { get; set; }

        public DateTime Creation_TimeStamp { get; set; }
        public Roles()
        {
            Id = Guid.NewGuid();
            IsDeleted = false;
            Creation_TimeStamp = DateTime.Now;
        }
    }


    public class SaveRole
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
    }
    public class LoadRole
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
    }
}

File: Client/src/app/model/Roles.ts
export interface RolesInterface {
  id?: number;
  name: string;
  isDeleted?: boolean;
}

export class Roles implements RolesInterface {
  id?: number;
  name: string;
  isDeleted?: boolean;

  constructor(obj: RolesInterface) {
    this.id = obj.id;
    this.name = obj.name;
    this.isDeleted = obj.isDeleted;
  }
}
